src/ast/visitor.rs

```diff
@@ -266,8 +266,8 @@ pub fn walk_expr<'a, V: Visitor<'a> + ?Sized>(visitor: &mut V, expr: &'a Expr) {
             }
         }
         ExprKind::NamedExpr { target, value } => {
-            visitor.visit_expr(target);
             visitor.visit_expr(value);
+            visitor.visit_expr(target);
```

Comment:
This ensures that when we read `exponential` on the RHS, we treat it as a usage of `exponential` from the parent scope.